### PR TITLE
[PF-2025] Add azure-unit-test; remove unusaed 'azure' profile

### PR DESCRIPTION
**Pull request by ddietterich**
The new controller test Doug added showed up a problem with the test categories. The new test needs the MockMvc to test the REST API interface. The `azure` tag is set up for a _connected_ test, so it does not support the MockMvc.

The default for `azure-enabled` is true, but we still need support for `azure-enabled` as false. So we still need to segregate the Azure tests.

My solution in this PR is:
1. Add a new tag `azure-unit`
2. Add a new profile `azure-unit-test`
3. Add a new property file that sets `azure-enabled`:  `application-azure-unit-test`
4. Added a fourth entry to the test matrix so these tests run as part of PR qualification as well as overnight.

I removed all of the `@ActiveProfiles("azure")` from a bunch of Azure tests. There is no profile-based code or configuration. The tests run fine without it.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        gradleTask: [unitTest, connectedTest, azureTest]
```
Comment (Qi77Qi) [inline .github/workflows/test.yml:41]: jsut for my understanding...what's the difference between `azureUnitTest` and `azureTest`?

**Review by jmczerk — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (ddietterich) [reply .github/workflows/test.yml:41]: `azureTest` is set up for connected tests with Sam, RBS, etc.  But it is not set up for using the MockMvc.

File: service/gradle/testing.gradle [hunk 2]
```diff
@@ -105,3 +105,29 @@ task azureTest(type: Test) {
...
+  useJUnitPlatform {
+    includeTags "azure-unit"
+  }
+  outputs.upToDateWhen { false }
+  finalizedBy tasks.combinedJaCoCoReport
+
+  // For GHA runs, retry failing tests. If 2nd time passes, workflow will still
+  // fail and oncall will still be notified. This way, oncall dosen't have to
+  // rerun workflow in GHA.
+  retry {
```
Comment (yuhuyoyo) [inline service/gradle/testing.gradle:119]: its unit test. do we need retry?

**Review by yuhuyoyo — APPROVED** [review]

File: service/gradle/testing.gradle (hunk 2 above)
Comment (ddietterich) [reply service/gradle/testing.gradle:119]: No, I don't think so. I was just cutting and pasting. Will remove.

**Review by Qi77Qi — APPROVED** [review]
would be nice to have some comments around `azureTest` vs `azureUnitTest` (or rename `azureTest` to `azureConnectedTest`) ?

File: .github/workflows/test.yml (hunk 1 above)
Comment (rtitle) [reply .github/workflows/test.yml:41]: Maybe we should `s/azureTest/azureConnectedTest` for clarity?

So basically:
* unitTest -- non-Azure unit tests
* connectedTest -- non-Azure connected tests
* azureUnitTest -- Azure unit tests only
* azureConnectedTest -- Azure connected tests only

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/CreateAzureDiskStepTest.java [hunk 3]
```diff
@@ -31,9 +31,7 @@
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
-import org.springframework.test.context.ActiveProfiles;
 
-@ActiveProfiles("azure")
 public class CreateAzureDiskStepTest extends BaseAzureTest {
```
Comment (rtitle) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/CreateAzureDiskStepTest.java:35]: I _think_ all the Azure `*StepTest`s can be `BaseAzureUnitTest`. (happy to do that in a separate PR too)

**Review by rtitle — APPROVED** [review]
Couple optional comments, but approving as is

**Comment by ddietterich** [comment]
Merging as is. I'll make a follow-up to improve documentation and rename for consistency.
